### PR TITLE
chore(litmus-portal): Home Page Stats for Litmus Portal.

### LINKED ISSUE #1788: Basic home screen analytics section UI development for Litmus Portal.
Basic home screen analytics section UI development for Litmus Portal.

### PR DESCRIPTION
**Pull request by ishangupta-ds**
## Proposed changes

Adding basic analytics for home page of litmus portal.
fixes #1788 

This PR also removes two outdated test cases from the cypress test suite of PassedVsFailed component.

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Dependency
- None.

## Special notes for your reviewer:
Analytics PR #1

### CODE REVIEW

File: litmus-portal/backend/auth/pkg/types/const.go [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 
 // define the type of authorization request
 var (
-	DefaultAPISecret           string = os.Getenv("JWT_SECRET")
-	DefaultUserName            string = os.Getenv("ADMIN_USERNAME")
-	DefaultUserPassword        string = os.Getenv("ADMIN_PASSWORD")
-	DefaultDBServerURL         string = os.Getenv("DB_SERVER")
+	DefaultAPISecret           string = "litmus-portal@123"
+	DefaultUserName            string = "admin"
+	DefaultUserPassword        string = "litmus"
```
Comment (arkajyotiMukherjee) [inline litmus-portal/backend/auth/pkg/types/const.go]: Can you change these back

File: litmus-portal/frontend/src/components/ProgressBar/AnalyticsLinearProgressBar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+const AnalyticsLinearProgressBar: React.FC<LinearProgressBarProps> = ({
+  value,
+  maxValue,
+}) => {
+  const [color, setColor] = useState(' ');
+  const width: number = 2;
+  const resultValue = ((value as number) / (maxValue as number)) * 100;
+  useEffect(() => {
+    return setColor('#5B44BA');
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/ProgressBar/AnalyticsLinearProgressBar/index.tsx:17]: why does color require to be a state variable if I never changes.
use a fixed color from theme data with useTheme()

File: litmus-portal/frontend/src/components/ProgressBar/AnalyticsLinearProgressBar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+const AnalyticsLinearProgressBar: React.FC<LinearProgressBarProps> = ({
+  value,
+  maxValue,
+}) => {
+  const [color, setColor] = useState(' ');
+  const width: number = 2;
+  const resultValue = ((value as number) / (maxValue as number)) * 100;
+  useEffect(() => {
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/ProgressBar/AnalyticsLinearProgressBar/index.tsx:16]: This useEffect is not needed

**Comment by ishangupta-ds** [comment]
image

File: litmus-portal/frontend/src/views/Home/PassedVsFailed/index.tsx [hunk 4]
```diff
@@ -58,12 +67,18 @@ const PassedVsFailed: React.FC<PassedVsFailedProps> = ({ passed, failed }) => {
             {failedValue === 0 ? (
               <div />
             ) : (
-              <img
-                src="./icons/Fail.png"
-                alt="Failed Icon"
+              <Avatar
+                style={{
+                  backgroundColor: '#CA2C2C',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/views/Home/PassedVsFailed/index.tsx:72]: put all colors in theme...if it is in react component use "useTheme" if it is used in styles...use normal theme value you get

**Review by arkajyotiMukherjee — CHANGES_REQUESTED** [review]

File: litmus-portal/backend/auth/pkg/types/const.go (hunk 1 above)
Comment (ishangupta-ds) [reply litmus-portal/backend/auth/pkg/types/const.go]: already reverted. @arkajyotiMukherjee

File: litmus-portal/frontend/src/components/ProgressBar/AnalyticsLinearProgressBar/index.tsx (hunk 2 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/ProgressBar/AnalyticsLinearProgressBar/index.tsx:17]: Some components were already there just reused code from them, will make changes for this.. @arkajyotiMukherjee

**Comment by ishangupta-ds** [comment]
@rajdas98 @arkajyotiMukherjee @amityt @S-ayanide PTAL.

File: litmus-portal/frontend/package.json [hunk 5]
```diff
@@ -17,6 +17,7 @@
     "@material-ui/core": "^4.10.2",
     "@material-ui/icons": "^4.9.1",
     "@material-ui/pickers": "^3.2.10",
+    "@types/lodash": "^4.14.161",
```
Comment (S-ayanide) [inline litmus-portal/frontend/package.json:20]: I believe this should be a dev dependency

File: litmus-portal/frontend/src/views/Home/AverageResilienceScore/index.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+import useStyles from './styles';
+
+interface AverageResilienceScoreProps {
+  value: number;
+}
+
+/*
+  Reusable Semi Circular Progress Bar
+  Required Params: value
+*/
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/views/Home/AverageResilienceScore/index.tsx:15]: Remove this comment, doesn't make sense in this context

**Review by S-ayanide — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/cypress/component/passedVsFailed.spec.tsx [hunk 7]
```diff
@@ -14,20 +14,6 @@ describe('Passed Vs Failed: props -> passed: 75, failed: 25', () => {
     expect(wrapper.props.failed).to.equal(25);
   });
 
-  it('Pass Icon is present', () => {
-    mount(wrapper);
-    cy.get('[data-cy=passIcon]')
-      .should('have.attr', 'src')
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/cypress/component/passedVsFailed.spec.tsx:20]: did we plan on removing all test cases @Jonsy13 @S-ayanide

**Comment by S-ayanide** [comment]
This PR is looks good to me, just a very minor change. :+1:

File: litmus-portal/frontend/cypress/component/passedVsFailed.spec.tsx (hunk 7 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/cypress/component/passedVsFailed.spec.tsx:20]: yeah we are not using these icons i am using the available icons in material ui so these conditions will always fail @arkajyotiMukherjee cc: @Jonsy13 @S-ayanide

File: litmus-portal/frontend/src/views/Home/ReturningHome/index.tsx [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+  );
+
+  const userData = useSelector((state: RootState) => state.userData);
+
+  // Query to get workflows
+  const { data, loading, error } = useQuery<Workflow, WorkflowDataVars>(
+    WORKFLOW_DETAILS,
+    {
+      variables: { projectID: userData.selectedProjectID },
+      fetchPolicy: 'cache-and-network',
```
[the remark this answers is not in the document]
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/views/Home/ReturningHome/index.tsx]: Removed @arkajyotiMukherjee

File: litmus-portal/frontend/cypress/component/passedVsFailed.spec.tsx (hunk 7 above)
Comment (Jonsy13) [reply litmus-portal/frontend/cypress/component/passedVsFailed.spec.tsx:20]: No @arkajyotiMukherjee It's not decided.Tests will remain.

Comment (ishangupta-ds) [reply litmus-portal/frontend/cypress/component/passedVsFailed.spec.tsx:20]: without icons how do i make the old test cases pass? @Jonsy13 @arkajyotiMukherjee

Comment (S-ayanide) [reply litmus-portal/frontend/cypress/component/passedVsFailed.spec.tsx:20]: @ishangupta-ds Can we add the two tests but instead use you Avatar icon and other icons from MUI itself? If that's possible for now!

Comment (Jonsy13) [reply litmus-portal/frontend/cypress/component/passedVsFailed.spec.tsx:20]: Yes @S-ayanide That should be good.

**Comment by ishangupta-ds** [comment]
@arkajyotiMukherjee @S-ayanide @Jonsy13 @rajdas98 PTAL.

**Review by S-ayanide — APPROVED** [review]

**Review by arkajyotiMukherjee — APPROVED** [review]

File: litmus-portal/frontend/cypress/component/passedVsFailed.spec.tsx (hunk 7 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/cypress/component/passedVsFailed.spec.tsx:20]: It is not accessible as png now we have to check for the material icon, i have another PR pending for the analytics, will be adding test cases after this release @S-ayanide @Jonsy13 ..if it is good?

**Review by Jonsy13 — APPROVED** [review]

**Pull request merged**